front/pages/w/[wId]/ds/[name]/upsert.tsx

```diff
@@ -161,6 +166,39 @@ export default function DataSourceUpsert({
     setTags(newTags);
   };
 
+  const checkUrl = (
+    value: string
+  ): { isValid: boolean; processedUrl: string | null } => {
+    let url = value.trim();
+    if (!isValidUrl(url)) {
+      return { isValid: false, processedUrl: null };
+    }
+    if (!url.startsWith("http://") && !url.startsWith("https://")) {
+      url = "http://" + url;
```

Comment:
https by default maybe?

Reply:
I thiiiiink we should handle that in the UI to avoid any surprise and make sure there's a protocol by API?

Reply:
> I thiiiiink we should handle that in the UI to avoid any surprise and make sure there's a protocol by API?

WDYM ? Reject if there is no `http[s]://` at the start ? or automatically insert it so its visible ?

Alternatively, I could let the front end send the URL as-is (without http/https) and have the backend (next) try with https and use that if it works

Reply:
The conservative approach is to reject. Automatically adding seems OK as well, but if we think about how these will be used I think rejecting makes sense (you're really supposed to copy your browser URL here)

Reply:
I presume `URL` does reject if there is no scheme/protocol right?

Reply:
Yes it does

Reply:
As discussed, I got rid of the UI part entirely